src/models/mentors/actions.js

```diff
@@ -26,52 +46,111 @@ export const deleteMentor = (id) => ({
 
 function* fetchAsyncMentors() {
   try {
-    yield put({ type: types.LOADING_STARTED });
+    yield put({ type: types.FETCHING_MENTORS_STARTED });
     const mentors = yield call(ApiService.load, '/mentors');
-    yield put({ type: types.FETCHING_MENTORS_SUCCESS, payload: { mentors } });
+    yield put({ type: types.FETCHING_MENTORS_SUCCEED, payload: { mentors } });
+  } catch (error) {
+    yield put({ type: types.FETCHING_MENTORS_FAILED, payload: { error } });
+  }
+}
+
+function* fetchAsyncMentorById({ payload }) {
+  try {
+    yield put({ type: types.FETCHING_BY_ID_STARTED });
+    const mentorId = payload.id;
+    const mentor = yield call(ApiService.load, `/mentors/${mentorId}`);
+    yield put({ type: types.FETCHING_BY_ID_SUCCEED, payload: { mentor } });
+  } catch (error) {
+    yield put({ type: types.FETCHING_BY_ID_FAILED, payload: { error } });
+  }
+}
+
+function* fetchAsyncActiveMentors() {
+  try {
+    yield put({ type: types.FETCHING_ACTIVE_STARTED });
+    const mentorsActive = yield call(ApiService.load, '/mentors/active');
+    yield put({ type: types.FETCHING_ACTIVE_SUCCEED, payload: { mentorsActive } });
+  } catch (error) {
+    yield put({ type: types.FETCHING_ACTIVE_SUCCEED, payload: { error } });
+  }
+}
+
+function* fetchAsyncMentorsGroups({ payload }) {
+  try {
+    yield put({ type: types.FETCHING_MENTOR_GROUPS_STARTED });
+    const mentorId = payload.id;
+    const mentorGroups = yield call(ApiService.load, `/mentors/${mentorId}/groups`);
+    yield put({ type: types.FETCHING_MENTOR_GROUPS_SUCCEED, payload: { mentorGroups } });
+  } catch (error) {
+    yield put({ type: types.FETCHING_MENTOR_GROUPS_FAILED, payload: { error } });
+  }
+}
+
+function* fetchAsyncMentorsCourses({ payload }) {
+  try {
+    yield put({ type: types.FETCHING_MENTOR_COURSES_STARTED });
+    const mentorId = payload.id;
+    const mentorCourses = yield call(ApiService.load, `/mentors/${mentorId}/courses`);
+    yield put({ type: types.FETCHING_MENTOR_COURSES_SUCCEED, payload: { mentorCourses } });
```

Comment:
```
const data = yield call(ApiService.load, `/mentors/${mentorId}/courses`);
yield put({ type: types.FETCHING_MENTOR_COURSES_SUCCEED, payload: { data } });
```
Because in reducer you use `action.payload.data`:
```
 case types.FETCHING_MENTOR_COURSES_SUCCEED:
      return {
        ...state,
        data: action.payload.data,
        isLoading: false,
        isLoaded: true,
      };
```